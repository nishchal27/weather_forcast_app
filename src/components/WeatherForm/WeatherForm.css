.WeatherForm {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
}

.WeatherForm > button,
.WeatherForm > input {
  width: 100%;
}

.WeatherForm > input {
  display: block;
  font-size: 1em;
  padding: 0.5rem 1rem;
  border: 2px solid var(--primary-color);
  border-radius: 5px;
  color: #333;
  transition: all 0.3s ease-out;
  margin-bottom: 10px;
}

.WeatherForm > input:hover {
  border-radius: 8px;
}

.WeatherForm > input:focus {
  outline: none;
  border-radius: 8px;
  border-color: var(--secondary-color);
}

.WeatherForm > button {
  display: block;
  padding: 0.5rem 1rem;
  font-size: 1em;
  color: white;
  background-color: var(--primary-color);
  border: 2px solid var(--primary-color);
  border-radius: 5px;
  transition: all 0.3s ease-out;
}

.WeatherForm > button:hover {
  border-radius: 8px;
}

.WeatherForm > button:focus {
  outline: none;
  border-radius: 8px;
  border-color: var(--secondary-color);
  background-color: var(--secondary-color);
}

@media (min-width: 600px) {
  .WeatherForm > button,
  .WeatherForm > input {
    width: auto;
  }

  .WeatherForm > input {
    margin-bottom: 0;
    margin-right: 10px;
  }
}
