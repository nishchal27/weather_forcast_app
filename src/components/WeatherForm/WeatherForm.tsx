import * as React from 'react';

import './WeatherForm.css';

type Props = {
  city: string,
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
  onSubmit: (e: React.FormEvent) => void,
};

const WeatherForm: React.FC<Props> = ({
  city,
  onChange,
  onSubmit,
}) => (
  <form className='WeatherForm' onSubmit={onSubmit}>
    <input
      name='city'
      aria-label='City'
      value={city}
      placeholder='Your city name'
      onChange={onChange}
    />
    <button
      type='submit'
      onSubmit={onSubmit}
    >
      Search
    </button>
  </form>
);

export default WeatherForm;
