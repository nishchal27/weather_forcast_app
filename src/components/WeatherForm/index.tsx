import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { WeatherAction } from '../../types/actions';
import { getWeather } from '../../actions';
import WeatherForm from './WeatherForm';

type State = {
  city: string;
};

type Props = {
  getWeather: (city: string) => WeatherAction;
};

export class WeatherFormContainer extends React.Component<Props, State> {
  state: State = {
    city: '',
  };

  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      city: e.target.value,
    });
  };

  handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    this.props.getWeather(this.state.city);
  };

  render() {
    return (
      <WeatherForm
        city={this.state.city}
        onChange={this.handleInputChange}
        onSubmit={this.handleSubmit}
      />
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  getWeather: bindActionCreators(getWeather, dispatch),
});

export default connect(
  null,
  mapDispatchToProps,
)(WeatherFormContainer);
