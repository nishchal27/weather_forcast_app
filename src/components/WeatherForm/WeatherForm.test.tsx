import React from 'react';
import { shallow } from 'enzyme';

import WeatherForm from './WeatherForm';

describe('<WeatherForm />', () => {
  it('renders without crashing', () => {
    const city = 'Madrid';
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const wrapper = shallow(
      <WeatherForm
        city={city}
        onChange={onChange}
        onSubmit={onSubmit}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
