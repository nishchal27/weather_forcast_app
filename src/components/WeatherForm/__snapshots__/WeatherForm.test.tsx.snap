// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherForm /> renders without crashing 1`] = `
<form
  className="WeatherForm"
  onSubmit={[MockFunction]}
>
  <input
    aria-label="City"
    name="city"
    onChange={[MockFunction]}
    placeholder="Your city name"
    value="Madrid"
  />
  <button
    onSubmit={[MockFunction]}
    type="submit"
  >
    Search
  </button>
</form>
`;
